File: Algorithms/Java/Swap Nodes in Pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
    	ListNode curr = new ListNode(-1);
		curr.next = head;
		head = curr;
		while (curr.next != null && curr.next.next != null) {
			ListNode temp = curr.next.next.next;
			curr.next.next.next = curr.next;
			curr.next = curr.next.next;
			curr.next.next.next = temp;
			curr = curr.next.next;
		}
		return head.next; 
    }
}

File: Algorithms/Python/Swap-Nodes-in-Pairs.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution(object):

    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head or not head.next:
            return head
        dummy = ListNode(-1)
        dummy.next = head
        node = dummy
        while node and node.next and node.next.next:
            slow = node.next
            fast = node.next.next
            node.next = fast
            slow.next = fast.next
            fast.next = slow
            node = node.next.next
        return dummy.next
